test/integration/scheduler/queue_test.go

```diff
@@ -371,6 +371,122 @@ func TestCoreResourceEnqueue(t *testing.T) {
+					return fmt.Errorf("failed to update the pod: %w", err)
+				}
+				return nil
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.SchedulerQueueingHints, true)
+
+			// Use zero backoff seconds to bypass backoffQ.
+			// It's intended to not start the scheduler's queue, and hence to
+			// not start any flushing logic. We will pop and schedule the Pods manually later.
+			testCtx := testutils.InitTestSchedulerWithOptions(
+				t,
+				testutils.InitTestAPIServer(t, "core-res-enqueue", nil),
+				0,
+				scheduler.WithPodInitialBackoffSeconds(0),
+				scheduler.WithPodMaxBackoffSeconds(0),
+			)
+			testutils.SyncSchedulerInformerFactory(testCtx)
+
+			defer testCtx.Scheduler.SchedulingQueue.Close()
+
+			cs, ns, ctx := testCtx.ClientSet, testCtx.NS.Name, testCtx.Ctx
+			// Create initialNode.
+			if _, err := cs.CoreV1().Nodes().Create(ctx, tt.initialNode, metav1.CreateOptions{}); err != nil {
+				t.Fatalf("Failed to create an initial Node %q: %v", tt.initialNode.Name, err)
+			}
+
+			if tt.initialPod != nil {
+				if _, err := cs.CoreV1().Pods(ns).Create(ctx, tt.initialPod, metav1.CreateOptions{}); err != nil {
+					t.Fatalf("Failed to create an initial Pod %q: %v", tt.initialPod.Name, err)
+				}
+			}
+
+			for _, pod := range tt.pods {
+				if _, err := cs.CoreV1().Pods(ns).Create(ctx, pod, metav1.CreateOptions{}); err != nil {
+					t.Fatalf("Failed to create Pod %q: %v", pod.Name, err)
+				}
+			}
+
+			// Wait for the tt.pods to be present in the scheduling queue.
+			if err := wait.PollUntilContextTimeout(ctx, time.Millisecond*200, wait.ForeverTestTimeout, false, func(ctx context.Context) (bool, error) {
+				pendingPods, _ := testCtx.Scheduler.SchedulingQueue.PendingPods()
+				return len(pendingPods) == len(tt.pods), nil
+			}); err != nil {
+				t.Fatal(err)
+			}
+
+			t.Log("Confirmed Pods in the scheduling queue, starting to schedule them")
+
+			// Pop all pods out. They should be unschedulable.
+			for i := 0; i < len(tt.pods); i++ {
+				testCtx.Scheduler.ScheduleOne(testCtx.Ctx)
+			}
+			// Wait for the tt.pods to be still present in the scheduling queue.
+			if err := wait.PollUntilContextTimeout(ctx, time.Millisecond*200, wait.ForeverTestTimeout, false, func(ctx context.Context) (bool, error) {
+				pendingPods, _ := testCtx.Scheduler.SchedulingQueue.PendingPods()
+				return len(pendingPods) == len(tt.pods), nil
+			}); err != nil {
+				t.Fatal(err)
+			}
+
+			t.Log("finished initial schedulings for all Pods, will trigger triggerFn")
+
+			err := tt.triggerFn(testCtx)
+			if err != nil {
+				t.Fatalf("Failed to trigger the event: %v", err)
+			}
+
+			t.Log("triggered tt.triggerFn, will check if tt.requeuedPods are requeued")
+
+			// wait some time to let scheduling_queue handle the triggered event
+			// TODO: any more elegant way ?
+			time.Sleep(time.Second * 10)
```

Comment:
@alculquicondor @sanposhiho 
I added integration tests for `tainttoleration` and `noderesources` in second commit.
And I think we also need a test to make sure that pods won't get moved to active Q by irrelevant events. So I created this mirror test.
But, is there an elegant way to know that schedulingQueue/schedulerCache has already processed the event? Currently I'm just using sleep here. It can be a waste of time sometimes.

Reply:
> I think we also need a test to make sure that pods won't get moved to active Q by irrelevant events.

Is it difficult to improve `TestCoreResourceEnqueue` to accommodate such scenarios, instead of having a separate test?

> is there an elegant way to know that schedulingQueue/schedulerCache has already processed the event?

Well, that's a good point.. (I'm thinking 🤔 )

Reply:
> Is it difficult to improve TestCoreResourceEnqueue to accommodate such scenarios too, instead of having a separate test?

Yes, I think improve `TestCoreResourceEnqueue` will make the test logic way too much complicate.

For `TestCoreResourceEnqueue`, we can use the same logic with QHint enabled/disabled. Because any event that makes a pod schedulable with QHint disabled can always make a pod schedulable with QHint enabled.

However, for `TestCoreResourceNOEnqueue`. A event that makes a pod schedulable with QHint disabled may can't make a pod schedulable with QHint enabled. So we can't run the same logic with QHint enabled/disabled twice.

And I think that as each plugin adds its logic to the tests, the testing logic can become more complex. I prefer writing two separate tests to make the logic of each test more maintainable.

Reply:
Yeah, we definitely wouldn't merge a plain sleep.

> is there an elegant way to know that schedulingQueue/schedulerCache has already processed the event?

Maybe it's worth adding a metric?

Reply:
> Maybe it's worth adding a metric?

Maybe leave this as a follow up?

Reply:
For now, can we call `scheduler.SchedulingQueue.Update` directly in triggerFn instead of actual Pod updating? That's much better than sleep.

Metric is an ideal way, which we can leave as a follow up; we can eventually replace triggerFn with actual Pod updating + check with metric in the future.

Reply:
> call scheduler.SchedulingQueue.Update directly in triggerFn

Nice idea! 😂